gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagProcessingEngine.java

```diff
@@ -50,31 +52,51 @@
 @Alpha
 @Slf4j
 @Singleton
-public class DagProcessingEngine {
+@AllArgsConstructor
+public class DagProcessingEngine extends AbstractIdleService {
 
   @Getter private final Optional<DagTaskStream> dagTaskStream;
   @Getter Optional<DagManagementStateStore> dagManagementStateStore;
+  private final Config config;
+  private final Optional<DagProcFactory> dagProcFactory;
+  private ScheduledExecutorService scheduledExecutorPool;
+  private static final Integer TERMINATION_TIMEOUT = 30;
 
   @Inject
   public DagProcessingEngine(Config config, Optional<DagTaskStream> dagTaskStream,
       Optional<DagProcFactory> dagProcFactory, Optional<DagManagementStateStore> dagManagementStateStore) {
+    this.dagTaskStream = dagTaskStream;
+    this.dagManagementStateStore = dagManagementStateStore;
+    this.config = config;
+    this.dagProcFactory = dagProcFactory;
```

Comment:
given `startup` immediately calls `.get` on these three `Optional`s, let's use the ctor to verify none are absent.  e.g. throw if any is